src/java.base/share/classes/java/util/Random.java

```diff
@@ -80,8 +81,143 @@
         name = "Random",
         i = 48, j = 0, k = 0,
         equidistribution = 0
-)
+        )
 public class Random implements RandomGenerator, java.io.Serializable {
+
+    /**
+     * Class used to wrap a {@link java.util.random.RandomGenerator} to
+     * {@link java.util.Random}.
+     */
+
+    @SuppressWarnings("serial")
+    private static class RandomWrapper extends Random implements RandomGenerator {
+        private final RandomGenerator generator;
+        private final boolean initialized;
+
+        private RandomWrapper(RandomGenerator randomToWrap) {
+            this.generator = randomToWrap;
+            this.initialized = true;
+        }
+
+        public static Random wrap(RandomGenerator random) {
+            // Check to see if its not wrapping another Random instance
+            if (random instanceof Random rand)
+                return rand;
+
+            return (Random) new Random.RandomWrapper(random);
```

Comment:
Isn't the `(Random)` cast redundant?

Reply:
Eclipse did not complain about it being redundant, also it does not impact the reading of it and it shows the reader it is a Random instance that is returning

Reply:
It is unlikely that any IDE or code scanning tool will report _all_ cases of incorrect or redundant code. However, in this specific case Eclipse does support detecting that (to my knowledge) through its "Clean Up" action (you might to explicitly enable though). But you should always verify that the actions Eclipse (or any other IDE) performs are correct, because there are some reported corner cases where it removes casts which are actually needed.